app/src/main/java/edu/grinnell/appdev/grinnelldirectory/Activity/LoginActivity.java

```diff
@@ -35,9 +46,61 @@ protected void onCreate(Bundle savedInstanceState) {
      *
      * @param view LoginActivity's view
      */
-    @OnClick(R.id.signin)
-    public void signIn(View view) {
-        String usernameTxt = mUsernameEditText.getText().toString();
-        String passwordTxt = mPasswordEditText.getText().toString();
+    @OnClick(R.id.login)
+    void signIn(View view) {
+        String username = mUsernameEditText.getText().toString();
+        String password = mPasswordEditText.getText().toString();
+        User user = new User(username, password);
+        DBAPICaller dbapiCaller = new DBAPICaller(user, this);
+        dbapiCaller.authenticateUser();
+    }
+
+    @Override public void onSearchSuccess(List<Person> people) {}
+
+    /**
+     * Save credentials and move to MainActivity if login succeeded
+     *
+     * @param success whether the login was successful
+     * @param person model for the logged in user
+     */
+    @BindString(R.string.authentication_failure) String authenticationFailure;
+    @Override public void authenticateUserCallSuccess(boolean success, Person person) {
+        final String username = mUsernameEditText.getText().toString();
+        final String password = mPasswordEditText.getText().toString();
+        if (success) {
+            saveCredentials(this, username, password);
+            saveUserDetails(this, person);
+
+            final Intent intent = new Intent(this, MainActivity.class);
+            startActivity(intent);
+        } else {
+            showAlert(authenticationFailure);
+        }
+    }
+
+    /**
+     * Show an error message if the server returns an error
+     *
+     * @param fail_message error description
+     */
+    @BindString(R.string.server_failure) String serverFailure;
+    @Override public void onServerFailure(String fail_message) {
+        showAlert(serverFailure + fail_message);
+    }
+
+    /**
+     * Show an error message if the network has an error
+     *
+     * @param fail_message error description
+     */
+    @BindString(R.string.networking_error) String networkingError;
+    @Override public void onNetworkingError(String fail_message) {
+        showAlert(networkingError + fail_message);
+    }
+
+    private void showAlert(String message) {
+        final AlertDialog.Builder builder = new AlertDialog.Builder(this);
+        builder.setMessage(message);
```

Comment:
@as-f Like larry said, we can have two inputs, but have one as the title and one as the message maybe. Ex. the "Network Failure" can be the title, and the error message can be the message part.